src/formatDate.js

```diff
@@ -9,6 +9,54 @@
  */
 function formatDate(date, fromFormat, toFormat) {
   // write code here
+
+  const oldSeparator = fromFormat[fromFormat.length - 1];
+  const newSeparator = toFormat[toFormat.length - 1];
+
+  const oldDateParts = date.split(oldSeparator);
+
+  let datePartsMatched = {};
+
+  for (let i = 0; i < oldDateParts.length; i++) {
+    datePartsMatched[fromFormat[i]] = oldDateParts[i];
+  }
+
+  datePartsMatched = convertYear(datePartsMatched);
+
+  const newDateParts = [];
+
+  for (let i = 0; i < oldDateParts.length; i++) {
+    const newDateFormat = toFormat[i];
+
+    const newDatePart = datePartsMatched[newDateFormat];
+
+    newDateParts.push(newDatePart);
+  }
```

Comment:
The loop iterates based on the length of `oldDateParts`, but it should iterate based on the length of `toFormat` minus one (to exclude the separator). This ensures that we're constructing the new date format correctly even if the `fromFormat` and `toFormat` arrays have different lengths.